packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedPacketHistroyDetails.tsx

```diff
@@ -32,57 +32,63 @@ const useStyles = makeStyles()((theme) => ({
         fontSize: 14,
         fontWeight: 700,
         lineHeight: '18px',
+        padding: '0 12px',
+    },
+    noData: {
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        width: '100%',
+        height: '318px',
+        fontSize: 14,
+        fontWeight: 700,
+        lineHeight: '24px',
+        color: theme.palette.maskColor.secondaryDark,
     },
 }))
 
 interface Props {
     rpid: string
 }
 
-export const FireflyRedPacketHistoryDetails = memo(function RedPacketPast({ rpid }: Props) {
+export const FireflyRedPacketHistoryDetails = memo(function FireflyRedPacketHistoryDetails({ rpid }: Props) {
     const { classes } = useStyles()
-    const {
-        data: claimData,
-        isLoading,
-        fetchNextPage,
-    } = useSuspenseInfiniteQuery<FireflyRedPacketAPI.RedPacketClaimListInfo>({
+    const t = useRedPacketTrans()
+    const { data: claimData, fetchNextPage } = useSuspenseInfiniteQuery<FireflyRedPacketAPI.RedPacketClaimListInfo>({
         queryKey: ['fireflyClaimHistory', rpid],
         initialPageParam: '',
         queryFn: async ({ pageParam }) => {
             const res = await FireflyRedPacket.getClaimHistory(rpid, createIndicator(undefined, pageParam as string))
             return res
         },
-        getNextPageParam: (lastPage) => lastPage.cursor,
+        getNextPageParam: (lastPage) => lastPage?.cursor,
     })
 
     const { claimInfo, claimList } = useMemo(
-        () => ({ claimList: claimData?.pages.flatMap((x) => x.list) ?? [], claimInfo: first(claimData?.pages) }),
+        () => ({ claimList: claimData?.pages.flatMap((x) => x?.list) ?? [], claimInfo: first(claimData?.pages) }),
         [claimData],
     )
 
-    if (isLoading) return <LoadingStatus className={classes.placeholder} iconSize={30} />
-
     return (
         <div className={classes.container}>
             {claimInfo ?
                 <FireflyRedPacketDetailsItem history={{ ...claimInfo, redpacket_id: rpid }} />
             :   null}
             <Box>
-                {claimList.map((item) => (
-                    <div className={classes.claimer} key={item.creator}>
-                        <Box>
+                {claimList.length ?
+                    claimList.map((item) => (
+                        <div className={classes.claimer} key={item?.creator}>
```

Comment:
Please use `compact` from `lodash-es` to filter nullish items out.

```suggestion
                        <div className={classes.claimer} key={item.creator}>
```